File: src/main/java/utils/java/ListNode.java
package utils.java;

public class ListNode {
    public int val;
    public ListNode next;

    public ListNode(int x) {
        val = x;
        next = null;
    }

    @Override
    public String toString() {
        return val + (next == null ? "" : "->" + next.toString());
    }

    public static ListNode parse(String str) {
        ListNode head = new ListNode(-1);
        ListNode cur = head;
        for (String s : str.split("->")) {
            cur.next = new ListNode(Integer.valueOf(s));
            cur = cur.next;
        }
        return head.next;
    }
}

File: src/main/java/utils/ListNode.kt
package utils

class ListNode(var `val`: Int, var next: ListNode? = null) {

    companion object {
        fun parse(str: String): ListNode? {
            val head = ListNode(-1)
            var cur = head
            str.split("->").map { it.toInt() }.forEach {
                cur.next = ListNode(it)
                cur = cur.next!!
            }
            return head.next
        }
    }

    override fun toString(): String {
        return "$`val`->$next"
    }
}